charts/kong/templates/_helpers.tpl

```diff
@@ -535,7 +535,17 @@ The name of the service used for the ingress controller's validation webhook
   {{- toYaml .Values.resources | nindent 4 }}
 {{- end -}}
 
+{{/*
+kong.controller2xplus returns "true" if the controller image tag major version is >= 2
+Note that this is a string, not a boolean, because templates vov
+*/}}
+{{- define "kong.controller2xplus" -}}
```

Comment:
consider an alternative that would be resilient to the `v` prefix being present or not:

`regexMatch "^v?2\." .Values.ingressController.image.tag`

(based on an understanding that the implementation using `semver` is more restrictive)

Reply:
It's at least tolerant of that:

```
$ helm template ex /tmp/symkong --set serviceMonitor.enabled=true --set ingressController.image.tag=v2.2 | grep -A2 cmetrics 
        - name: cmetrics
          containerPort: 10255
          protocol: TCP

09:56:21-0700 yagody $ helm template ex /tmp/symkong --set serviceMonitor.enabled=true --set ingressController.image.tag=v1.2 | grep -A2 cmetrics 

$ helm template ex /tmp/symkong --set serviceMonitor.enabled=true --set ingressController.image.tag=dfgdfgv1.2 | grep -A2 cmetrics 
Error: template: kong/templates/deployment.yaml:93:10: executing "kong/templates/deployment.yaml" at <include "kong.controller-container" .>: error calling include: template: kong/templates/_helpers.tpl:548:27: executing "kong.controller-container" at <include "kong.controller2xplus" .>: error calling include: template: kong/templates/_helpers.tpl:543:16: executing "kong.controller2xplus" at <semver .Values.ingressController.image.tag>: error calling semver: Invalid Semantic Version

Use --debug flag to render out invalid YAML
```